client.go

```diff
@@ -1375,3 +1398,455 @@ func (c *ToznySDKV3) UnbrokerShare(ctx context.Context, authorizerClientID strin
 		RecordType:   recordType,
 	})
 }
+
+type SecretRequest struct {
+	SecretType  string
+	SecretName  string
+	SecretValue string
+	Description string
+	FileName    string
+}
+
+// CreateSecret makes a secret of the specified type and shares it with the client
+func (c *ToznySDKV3) CreateSecret(ctx context.Context, login TozIDLoginRequest, secret SecretRequest) (*pdsClient.Record, string, error) {
+	var createdSecret *pdsClient.Record
+	SECRET_TYPES := []string{"Client", "Credential", "File"}
+	// actually need to check the input for secret here too
+	secret.SecretName = strings.TrimSpace(secret.SecretName)
+	secret.SecretValue = strings.TrimSpace(secret.SecretValue)
+	if secret.SecretType == "" {
+		return createdSecret, "type cannot be empty", nil
```

Comment:
the more idiomatic go way to do this is to not initialize the var on line 1412, and if you are returning early because of an error case, return nil for the Record return value. The go idiom is "if you call a function, and it returns an error and other values, always check the error first, and if the error is not nil then you should assume all the other values should not be used / aren't valid (like a nil pointer for example here)